pipeline/download_iem.py

```diff
@@ -0,0 +1,158 @@
+"""Download ASOS wind data for preprocessing and app ingest
+
+Currently hardwired to download all wind speed and direction data for 
+all AK_ASOS network stations, for 1980-2020
+"""
+
+import os, requests, time
+from multiprocessing import Pool
+from pathlib import Path
+
+
+BASE_URI = "http://mesonet.agron.iastate.edu/"
+
+
+def get_stations(network):
+    """Build a station list from a network
+
+    Args:
+        network (str): namne of IEM ASOS network to use (defaults to AK)
+
+    Returns:
+        a list of four letter ICAO station identifiers
+    """
+    uri = (
+        f"{BASE_URI}geojson/network/{network}.geojson"
+    )
+    r = requests.get(uri)
+    jdict = r.json()
+    return  [site["properties"]["sid"] for site in jdict["features"]]
+
+
+def make_uris(sids, start, end):
+    """Make the URIs for all stations
+    
+    Args:
+        sids (list): List of IACO station identifiers (str)
+        start (str): Starting date in YYYY-mm-dd
+        end (str): ending date in YYYY-mm-dd
+
+    Returns:
+        a list of URIs, one for each dataset
+    """
+    service = BASE_URI + "cgi-bin/request/asos.py?"
+    # speed (mph), direction, and gust
+    service += "data=sped&data=drct&data=gust_mph&"
+    # comma-separated (no debug header) no latlot or elev
+    service += "&tz=Etc/UTC&format=onlycomma&latlon=no&elev=no&"
+    # get the hourly "routine + SPECals" (report type 2)
+    # report type 1 is MADIS HFMETAR
+    service += "report_type=2&"
+    # add start and ending dates
+    start, end = start.split("-"), end.split("-")
+    service += f"year1={start[0]}&month1={start[1]}&day1={start[2]}&"
+    service += f"year2={end[0]}&month2={end[1]}&day2={end[2]}&"
```

Comment:
No need to change this, but if you weren't already, it's helpful to be aware of the capability of libraries [like urllib3](https://urllib3.readthedocs.io/en/latest/user-guide.html) to create requests using key/value pairs instead of gluing strings together.  It tends to save you some occasional woes.